From 158c45a9a2555afca5adcfcebfa8e2dd3ff75507 Mon Sep 17 00:00:00 2001
From: q66 <daniel@octaforge.org>
Date: Sat, 7 Mar 2020 05:00:24 +0100
Subject: [PATCH] depend on sha1_h in src/vulkan/overlay-layer

Without this, the dependency ordering gets messed up and during
highly parallel builds the git_sha1.h file gets generated later
than overlay-layer is compiled. That results in build errors like:

overlay.cpp:31:10: fatal error: git_sha1.h: No such file or directory
---
 src/vulkan/overlay-layer/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git src/vulkan/overlay-layer/meson.build src/vulkan/overlay-layer/meson.build
index 5b553d0..51e053e 100644
--- src/vulkan/overlay-layer/meson.build
+++ src/vulkan/overlay-layer/meson.build
@@ -38,7 +38,7 @@ vklayer_files = files(
 
 vklayer_mesa_overlay = shared_library(
   'VkLayer_MESA_overlay',
-  vklayer_files, overlay_spv,
+  [vklayer_files, sha1_h], overlay_spv,
   c_args : [c_vis_args, no_override_init_args, vulkan_wsi_args],
   cpp_args : [cpp_vis_args, vulkan_wsi_args],
   dependencies : [idep_vulkan_util, idep_mesautil, vulkan_wsi_deps, libimgui_core_dep, dep_dl],
-- 
2.25.1

